os/arch/arm/src/armv8-m/up_schedulesigaction.c

```diff
@@ -234,6 +243,11 @@ void up_schedule_sigaction(struct tcb_s *tcb, sig_deliver_t sigdeliver)
 #ifdef CONFIG_BUILD_PROTECTED
 			tcb->xcp.regs[REG_LR] = EXC_RETURN_PRIVTHR;
 #endif
+			if (tcb->xcp.regs[REG_EXC_RETURN] == EXC_RETURN_HANDLER) {
+				tcb->xcp.regs[REG_EXC_RETURN] = EXC_RETURN_PRIVTHR;
+			}
+
+			tcb->xcp.regs[REG_EXC_RETURN] = EXC_RETURN_PRIVTHR;
```

Comment:
We need to set the T bit here also?

Reply:
@abhishek-samsung Could you check @kishore-sn 's comment?

Reply:
@sunghan-chang , I have resolved @kishore-sn 's comment and have pushed the changes for armv8m. Changes for armv7m are pending. I will do them in a different PR as I am facing some build issues with qemu in v7m. And regarding verification, I have run kernel TCs on loadable and all of them passed. I have also tested with the lpwork usagefualt patch (https://github.com/ewoodev/TizenRT/commit/3da790126090b41ce0ad9126630d81856d0abf1e) and it didn't cause any crash on both flat and loadable.